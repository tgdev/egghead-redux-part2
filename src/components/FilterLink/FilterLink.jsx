import React from 'react';
import PropTypes from 'prop-types';

const FilterLink = ({ active, children, handleFilter }) => {
  if (active) {
    return <span>{children}</span>;
  }

  return (
    <button onClick={() => handleFilter()}>
      {children}
    </button>
  );
};

FilterLink.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.string.isRequired,
  handleFilter: PropTypes.func
};

export default FilterLink;
