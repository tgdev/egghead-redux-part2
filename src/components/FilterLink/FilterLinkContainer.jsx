import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { setVisibilityFilter } from 'actions/visibility-filter.actions';

import FilterLink from './FilterLink';

const mapStateToProps = (state, ownProps) => {
  return {
    active: ownProps.filter === state.visibilityFilter
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    handleFilter: () => {
      dispatch(setVisibilityFilter(ownProps.filter));
    }
  };
};

const FilterLinkContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(FilterLink);

FilterLinkContainer.propTypes = {
  filter: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
};

export default FilterLinkContainer;
