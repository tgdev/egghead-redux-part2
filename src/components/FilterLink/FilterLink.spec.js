import React from 'react';
import { shallow, mount } from 'enzyme';

import FilterLink from './FilterLink';

describe('FilterLink component', () => {

  const mockActive = true;
  const mockNotActive = false;


  let mockClickHandler = jest.fn();

  afterEach(() => {
    mockClickHandler.mockReset();
  });

  it('renders FilterLink correctly', () => {
    const wrapper = shallow(
      <FilterLink
        active={mockActive}
        handleFilter={mockClickHandler}>
        All
      </FilterLink>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with required children', () => {
    const wrapper = shallow(
      <FilterLink
        active={mockActive}
        handleFilter={mockClickHandler}>
        All
      </FilterLink>
    );
    expect(wrapper.children().length).toBe(1);
  });

  it('renders current filter differently to other filters', () => {
    const wrapperActive = shallow(
      <FilterLink
        active={mockActive}
        handleFilter={mockClickHandler}>
        All
      </FilterLink>
    );
    expect(wrapperActive).toMatchSnapshot();

    const wrapperInactive = shallow(
      <FilterLink
        active={mockNotActive}
        handleFilter={mockClickHandler}>
        All
      </FilterLink>
    );
    expect(wrapperInactive).toMatchSnapshot();
  });

  it('triggers handleToggleTodo when todo item is clicked', () => {
    const wrapper = mount(
      <FilterLink
        active={mockNotActive}
        handleFilter={mockClickHandler}>
        All
      </FilterLink>
    );
    wrapper.find('button').first().simulate('click');
    expect(mockClickHandler).toHaveBeenCalledTimes(1);
  });

});
