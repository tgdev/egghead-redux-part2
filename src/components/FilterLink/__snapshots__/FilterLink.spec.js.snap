// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLink component renders FilterLink correctly 1`] = `
<span>
  All
</span>
`;

exports[`FilterLink component renders current filter differently to other filters 1`] = `
<span>
  All
</span>
`;

exports[`FilterLink component renders current filter differently to other filters 2`] = `
<button
  onClick={[Function]}
>
  All
</button>
`;
